imports/components/@enhancers/likes/toggle.js

```diff
@@ -0,0 +1,82 @@
+
+// @flow
+
+// $FlowMeteor
+import { Meteor } from "meteor/meteor";
+import React, { Component } from "react";
+import { graphql } from "react-apollo";
+import gql from "graphql-tag";
+import { connect } from "react-redux";
+
+import {
+  nav as navActions,
+  liked as likedActions,
+  modal,
+} from "../../../data/store";
+
+import OnBoard from "../../people/accounts";
+
+export const TOGGLE_LIKE_MUTATION = gql`
+  mutation ToggleLike($nodeId: String!) {
+    toggleLike(nodeId: $nodeId) {
+      like {
+        id
+      }
+    }
+  }
+`;
+
+const withToggleLike = graphql(TOGGLE_LIKE_MUTATION, {});
+
+export const classWrapper = (propsReducer: Function) => (WrappedComponent: any) => {
+  type ILikesWrapper = {
+    dispatch: Function,
+    mutate: Function,
+  };
+
+  class LikesWrapper extends Component {
+    props: ILikesWrapper;
+
+    componentWillMount() {
+      this.props.dispatch(navActions.setLevel("CONTENT"));
+      this.props.dispatch(navActions.setAction("CONTENT", {
+        id: 2,
+        action: this.toggleLike,
+      }));
+    }
+
+    componentWillUnmount() {
+      this.props.dispatch(navActions.setLevel("TOP"));
+    }
+
+    getNodeId = () => {
+      if (propsReducer && typeof propsReducer === "function") {
+        return propsReducer(this.props);
+      }
+      console.warn("propsReducer was either not passed, or is not a function");
+      return null;
+    };
+
+    toggleLike = () => {
+      const { dispatch, mutate } = this.props;
+      if (!Meteor.userId()) { // if not logged in, show login modal
+        dispatch(modal.render(OnBoard, {
+          coverHeader: true, modalBackground: "light",
```

Comment:
@JakeDawkins lets add an `onFinishedAction` to like the entry after the user has logged in.

Check out https://github.com/NewSpring/Holtzman/blob/master/imports/components/giving/checkout-buttons/index.js#L122 for example